crates/cli_testing_examples/benchmarks/NQueens.roc

```diff
@@ -5,32 +5,31 @@ app "nqueens"
 
 main : Task.Task {} []
 main =
-    Task.after
-        Task.getInt
-        \n ->
+    inputResult <- Task.attempt Task.getInt
+
+    when inputResult is
+        Ok n ->
             queens n # original koka 13
-            |> Num.toStr
-            |> Task.putLine
+                |> Num.toStr
+                |> Task.putLine
+
+        Err GetIntError ->
+            Task.putLine "Error: Failed to get Integer from stdin."
 
 ConsList a : [Nil, Cons a (ConsList a)]
 
 queens = \n -> length (findSolutions n n)
 
-length : ConsList a -> I64
-length = \xs -> lengthHelp xs 0
+findSolutions = \n, k ->
+    if k <= 0 then # should we use U64 as input type here instead?
+        Cons Nil Nil
+    else
+        extend n Nil (findSolutions n (k - 1))
 
-lengthHelp : ConsList a, I64 -> I64
-lengthHelp = \foobar, acc ->
-    when foobar is
-        Cons _ lrest -> lengthHelp lrest (1 + acc)
+extend = \n, acc, solutions ->
+    when solutions is
```

Comment:
The order of functions was a bit weird to me here so I changed it to be descending based on importance.